web3/utils/filters.py

```diff
@@ -166,16 +161,51 @@ def format_entry(self, entry):
         return entry
 
     def set_data_filters(self, data_filter_set):
+        """(('uint256', [12345, 54321]), ('string', ('a-single-string',)))
+        """
         self.data_filter_set = data_filter_set
         if any(data_filter_set):
-            self.data_filter_set_regex = construct_data_filter_regex(
-                data_filter_set,
-            )
+            self.data_filter_set_function = match_fn(data_filter_set)
 
     def is_valid_entry(self, entry):
-        if not self.data_filter_set_regex:
+        if not self.data_filter_set:
             return True
-        return bool(self.data_filter_set_regex.match(entry['data']))
+        return bool(self.data_filter_set_function(entry['data']))
+
+
+def normalize_data_values(type_string, data_value):
```

Comment:
converting bytes to strings can go away with eth-abi v2

Reply:
We'll have to support old versions of eth-abi at least until web3.py v5, though.

Reply:
Also, this function probably deserves some small comments explaining what is happening and why.